windows.c

```diff
@@ -243,9 +257,23 @@ void setupSystemShutdownHandler()
   // Create an invisible window so that we can respond to system shutdown and
   // make sure that we finish setting the system proxy to off.
   DWORD tid;
-  HANDLE hInvisiblethread=CreateThread(NULL, 0, runInvisibleWindowThread, NULL, 0, &tid);
-  if (hInvisiblethread == NULL)
+  hInvisibleThread=CreateThread(NULL, 0, runInvisibleWindowThread, NULL, 0, &tid);
+  if (hInvisibleThread == NULL)
   {
     printf("FAILED to create thread for invisible window!!!  %Iu\n",GetLastError());
   }
+
+  // Make this process shut down earlier than parent so that in system logoff
+  // case, we successfully disable system proxy before parent process tried to
+  // do so and fails.
+  //
+  // See https://stackoverflow.com/questions/8760509/graceful-application-shutdown-when-windows-shuts-down
+  DWORD dwLevel, dwFlags;
+  BOOL fOkay = GetProcessShutdownParameters(&dwLevel, &dwFlags);
+  if (fOkay && dwLevel > 0x100) {
+    fOkay = SetProcessShutdownParameters(dwLevel + 1, SHUTDOWN_NORETRY);
+  }
+  if (!fOkay) {
+    printf("Failed to prioritize sysproxy-cmd for shutdown\n");
+  }
```

Comment:
The other change is that with this logic, we'll always shut down `sysproxy-cmd` before we shut down Lantern itself. That way, we avoid the possibility that Lantern somehow killed `sysproxy-cmd` before it had a chance to handle the `WM_ENDSESSION` event.